yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -86,6 +86,7 @@ private[spark] class YarnClientSchedulerBackend(
         ("--executor-cores", "SPARK_WORKER_CORES", "spark.executor.cores"),
         ("--executor-cores", "SPARK_EXECUTOR_CORES", "spark.executor.cores"),
         ("--queue", "SPARK_YARN_QUEUE", "spark.yarn.queue"),
+        ("--priority", "SPARK_YARN_PRIORITY", "spark.yarn.priority"),
```

Comment:
I'd really prefer to not add this. There's no need, just set the value in `SparkConf` and read it in `Client.scala` - which is what this line you're adding in SparkSubmit already does: 

    OptionAssigner(args.priority, YARN, CLIENT, sysProp = "spark.yarn.priority"),

Just make it work for both deploy modes.